compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/AggregatedOpInterfaceImpl.cpp

```diff
@@ -44,6 +44,71 @@ static Value scaleValueInPlace(OpBuilder &builder, Location loc,
   return genericOp.getResult(0);
 }
 
+static Value reciprocalValue(OpBuilder &b, Location loc, Value input,
+                             Value output) {
+  int64_t rank = cast<ShapedType>(input.getType()).getRank();
+  SmallVector<AffineMap> maps = {b.getMultiDimIdentityMap(rank),
+                                 b.getMultiDimIdentityMap(rank)};
+
+  SmallVector<utils::IteratorType> iteratorTypes(rank,
+                                                 utils::IteratorType::parallel);
+  auto genericOp = b.create<linalg::GenericOp>(
+      loc, output.getType(), ValueRange{input}, output, maps, iteratorTypes,
+      [&](OpBuilder &b, Location loc, ValueRange args) {
+        Value in = convertScalarToDtype(b, loc, args[0], args[1].getType(),
+                                        /*isUnsignedCast=*/false);
+        // Convert scale to the same datatype as input.
+        Value one =
+            b.create<arith::ConstantOp>(loc, b.getFloatAttr(in.getType(), 1.0));
+        Value result = b.create<arith::DivFOp>(loc, one, in);
+        b.create<linalg::YieldOp>(loc, result);
+      });
+  return genericOp.getResult(0);
+}
+
+static Value truncateFloat(OpBuilder &builder, Location loc, AffineMap inputMap,
+                           AffineMap outputMap, Value value, Value output) {
+  SmallVector<AffineMap> compressedMaps =
+      compressUnusedDims(SmallVector<AffineMap>{inputMap, outputMap});
+  inputMap = compressedMaps[0];
+  outputMap = compressedMaps[1];
+
+  SmallVector<utils::IteratorType> iteratorTypes(inputMap.getNumDims(),
+                                                 utils::IteratorType::parallel);
+  auto genericOp = builder.create<linalg::GenericOp>(
+      loc, output.getType(), value, output,
+      SmallVector<AffineMap>{inputMap, outputMap}, iteratorTypes,
+      [&](OpBuilder &b, Location loc, ValueRange args) {
+        auto srcTy = cast<FloatType>(args[0].getType());
+        auto dstTy = cast<FloatType>(args[1].getType());
+
+        // We clamp to the min / max of the floating point representation
+        double mnDbl =
+            APFloat::getLargest(dstTy.getFloatSemantics(), /*Negative=*/true)
+                .convertToDouble();
+        double mxDbl =
+            APFloat::getLargest(dstTy.getFloatSemantics(), /*Negative=*/false)
+                .convertToDouble();
+
+        Value mn = builder.create<arith::ConstantOp>(
+            loc, builder.getFloatAttr(srcTy, mnDbl));
+        Value mx = builder.create<arith::ConstantOp>(
+            loc, builder.getFloatAttr(srcTy, mxDbl));
+        Value gt = b.create<arith::CmpFOp>(loc, arith::CmpFPredicate::OGT,
+                                           args[0], mx);
+        Value lt = b.create<arith::CmpFOp>(loc, arith::CmpFPredicate::OLT,
+                                           args[0], mn);
+        Value sel0 = b.create<arith::SelectOp>(loc, gt, mx, args[0]);
+        Value sel1 = b.create<arith::SelectOp>(loc, lt, mn, sel0);
```

Comment:
Is all of this for numerical stability?

Reply:
Truncation to `fp8` requires checking that it remains in bounds. Even with the known max value / normalization it is possible that it exceeds the value and generates a `nan`. I need to check lower level compilation restrictions as I believe some configurations end up including the truncation.

Reply:
nice! :)

Reply:
NIT: I think this is quite cool and crucial info, can we add it as comment somewhere here? :)

Reply:
Added the comment.